src/main/java/com/exactpro/th2/check1/Checkpoint.java

```diff
@@ -97,27 +98,35 @@ public boolean equals(Object obj) {
 
         return new EqualsBuilder()
                 .append(id, other.id)
-                .append(sessionKeyToSequence, other.sessionKeyToSequence)
+                .append(sessionKeyToCheckpointData, other.sessionKeyToCheckpointData)
                 .isEquals();
     }
 
     @Override
     public int hashCode() {
         return new HashCodeBuilder(17, 37)
                 .append(id)
-                .append(sessionKeyToSequence)
+                .append(sessionKeyToCheckpointData)
                 .toHashCode();
     }
 
     public static Checkpoint convert(com.exactpro.th2.common.grpc.Checkpoint protoCheckpoint) {
-        Map<SessionKey, Long> sessionKeyToSequence = new HashMap<>();
+        Map<SessionKey, CheckpointData> sessionKeyToSequence = new HashMap<>();
         for (Map.Entry<String, DirectionCheckpoint> sessionAliasDirectionCheckpointEntry : protoCheckpoint.getSessionAliasToDirectionCheckpointMap().entrySet()) {
             String sessionAlias = sessionAliasDirectionCheckpointEntry.getKey();
             DirectionCheckpoint directionCheckpoint = sessionAliasDirectionCheckpointEntry.getValue();
-            for (Map.Entry<Integer, Long> directionSequenceEntry : directionCheckpoint.getDirectionToSequenceMap().entrySet()) {
-                SessionKey sessionKey = new SessionKey(sessionAlias, Direction.forNumber(directionSequenceEntry.getKey()));
-                sessionKeyToSequence.put(sessionKey, directionSequenceEntry.getValue());
-            }
+            if (directionCheckpoint.getDirectionToCheckpointDataCount() != 0) {
+                for (Map.Entry<Integer, com.exactpro.th2.common.grpc.Checkpoint.CheckpointData> directionSequenceEntry : directionCheckpoint.getDirectionToCheckpointDataMap().entrySet()) {
```

Comment:
Why don't you check if the gRPC checkpoint has only one map set? We should report an error if both old and new version are used